### PR TITLE
BHBC-1955: Summary Messages Update

### PR DESCRIPTION
**Pull request by AlfredRosenthal**
# Overview

## Description of relevant changes

- updated message type for missing header
- added check that survey species and template species line up properly

### Scenario 1
- Get a summary template upsert script from [biohubbc-utils](https://github.com/bcgov/biohubbc-utils)
- Open DBeaver and run the summary template scripts
- Start SIMS 
- download a summary template from the resources 
- Create Project/ Survey
- Open summary template from resources page and remove an entire column
- Upload modified file and see `Mandatory fields have not been filled` error

### Scenario 2
- Get a summary template upsert script from [biohubbc-utils](https://github.com/bcgov/biohubbc-utils)
- Open DBeaver and run the summary template scripts
- Start SIMS 
- download a summary template from the resources 
- Create Project/ Survey
- Upload a summary template intended for a different species than the one in the survey and see `Failed to get validation rules` error

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: api/src/repositories/summary-repository.ts [hunk 1]
```diff
@@ -428,6 +428,20 @@ export class SummaryRepository extends BaseRepository {
       throw failedToFindValidationRulesError;
     }
 
+    // check if the species in the survey and the species on the template match
+    if (species) {
+      if (!response.rows.some((row) => species.includes(Number(row.wldtaxonomic_units_id)))) {
```
Comment (curtisupshall) [inline api/src/repositories/summary-repository.ts:433]: Could be a good idea to write a couple tests covering these two branches

**Review by curtisupshall — DISMISSED** [review]

**Review by anissa-agahchen — DISMISSED** [review]

File: api/src/repositories/summary-repository.ts (hunk 1 above)
Comment (AlfredRosenthal) [reply api/src/repositories/summary-repository.ts:433]: good call, I've added some test around this code

**Comment by anissa-agahchen** [comment]
Looks good!

**Review by anissa-agahchen — APPROVED** [review]

**Review by KjartanE — APPROVED** [review]
🐈

**Pull request merged**